C++ API parity: AdaptiveMaxPool3d

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26915 C++ API parity: MaxUnpool2d
* #26896 C++ API parity: MaxUnpool1d
* #26819 C++ API parity: AdaptiveAvgPool3d
* #26818 C++ API parity: AdaptiveAvgPool2d
* #26808 C++ API parity: AdaptiveAvgPool1d
* **#26775 C++ API parity: AdaptiveMaxPool3d**

Differential Revision: [D17627824](https://our.internmc.facebook.com/intern/diff/D17627824)

pbelevich committed:
C++ API parity: AdaptiveMaxPool3d

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveMaxPool3d"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveMaxPool3d"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveMaxPool3d"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveMaxPool3d"




[ghstack-poisoned]

yf225 reviewed (approved):
Looks awesome, thanks @pbelevich !

yf225 reviewed (changes requested):
Oops sorry, seems like we are missing pretty print tests for `AdaptiveMaxPool3d`.

pbelevich committed:
Update on "C++ API parity: AdaptiveMaxPool3d"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveMaxPool3d"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveMaxPool3d"




[ghstack-poisoned]

yf225 reviewed (approved):
Thanks @pbelevich !

pbelevich committed:
Update on "C++ API parity: AdaptiveMaxPool3d"


Differential Revision: [D17627824](https://our.internmc.facebook.com/intern/diff/D17627824)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: AdaptiveMaxPool3d"


Differential Revision: [D17627824](https://our.internmc.facebook.com/intern/diff/D17627824)

[ghstack-poisoned]
